### PR TITLE
[ re #488 ] properties of Stream's map, repeat, and cycle

### PR DESCRIPTION
**Pull request by gallais**

### CODE REVIEW

**Comment by gallais** [comment]
Something that makes the library a bit hard to use: the `λ where .force → b` functions
are not equal to themselves. Or rather, they are not equal to another pattern-matching
lambda which is syntactically the same.

This is a well-known problem (although I can't seem to find an issue for it on the bug
tracker) but it is made quite visible by the current state of affairs in the stdlib: they
naturally show up in intermediate expressions when using equational reasoning. And
writing them down is a sure way for the proof not to go through. Cf. the multiple
sub-expressions I have left out in the proofs in this PR.

What should we do?

**Comment by JacquesCarette** [comment]
This problem seems to be one of agda, brought out by work on stdlib -- I would be inclined to post an issue directly on the agda/agda repo, flagging the main stakeholders for attention. Otherwise I'm not sure how much attention your comment on this PR is going to get!

**Comment by gallais** [comment]
I'm not sure anything can be done in Agda: I think the fact anonymous pattern
matching lambdas are not compared structurally is important to guarantee that
typechecking terminates.

One thing we can do on our side of the problem (if we care about users being
able to write down intermediate values in equational proofs) is to give names
to these anonymous pattern-matching lambdas. I'm not really fond of this idea
though tbh.

File: src/Data/List/Relation/Binary/Equality/Propositional.agda [hunk 1]
```diff
@@ -12,19 +12,22 @@ module Data.List.Relation.Binary.Equality.Propositional {a} {A : Set a} where
...
-≋⇒≡ []             = refl
-≋⇒≡ (refl ∷ xs≈ys) = cong (_ ∷_) (≋⇒≡ xs≈ys)
+≋⇒≡ []               = P.refl
+≋⇒≡ (P.refl ∷ xs≈ys) = P.cong (_ ∷_) (≋⇒≡ xs≈ys)
 
 ≡⇒≋ : _≡_ ⇒ _≋_
-≡⇒≋ refl = ≋-refl
+≡⇒≋ P.refl = ≋-refl
+
+refl : Reflexive _≋_
```
Comment (MatthewDaggitt) [inline src/Data/List/Relation/Binary/Equality/Propositional.agda]: This is exported as `≋-refl` already.

Comment (gallais) [reply src/Data/List/Relation/Binary/Equality/Propositional.agda]: Thanks!

File: CHANGELOG.md [hunk 2]
```diff
@@ -30,6 +30,27 @@ Data.List.Relation.Unary.Unique.Setoid.Properties
...
+  repeat-ap-identity ↦ ap-repeatˡ
+  ap-repeat-identity ↦ ap-repeatʳ
+  ```
+
+* Renamed a few lemmas in `Codata.Stream.Properties` to match the more stdlib
+  conventions:
+  ```agda
+  ap-repeat-commute  ↦ ap-repeat
+  map-repeat-commute ↦ map-repeat
+  ```
```
Comment (MatthewDaggitt) [inline CHANGELOG.md]: Is it possible to deprecate the old names? We should try to keep our promise of backwards-compatibility as much as possible!

**Comment by MatthewDaggitt** [comment]
Do `repeat-ap-identity` and `ap-repeat-identity` need deprecating too? Or am I missing something?

**Pull request merged**